services/user.py

```diff
@@ -0,0 +1,49 @@
+from django.contrib.auth import get_user_model
+
+
+def create_user(
+        username: str,
+        password: str,
+        email: str | None = None,
+        first_name: str | None = None,
+        last_name: str | None = None
+) -> get_user_model():
+    user = get_user_model().objects.create_user(
+        username=username,
+        password=password
+    )
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()
+
+    return user
+
+
+def get_user(user_id: int) -> get_user_model():
+    return get_user_model().objects.get(id=user_id)
+
+
+def update_user(
+        user_id: int,
+        username: str | None = None,
+        password: str | None = None,
+        email: str | None = None,
+        first_name: str | None = None,
+        last_name: str | None = None
+) -> None:
+    user = get_user(user_id)
+    if username:
+        user.username = username
+    if password:
+        user.set_password(password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()

```

Comment:
Try to use update function here

Reply:
In Django [documentation ](https://docs.djangoproject.com/en/5.1/topics/db/queries/) it is written:
_Sometimes you want to set a field to a particular value for all the objects in a QuerySet. You can do this with the update() method._
And these:
_To save changes to an object that’s already in the database, use save()._

I think in this case, because we update only 1 object, it is better to use save().

Also, with the update function I see two possible ways how to implement it here and problems that arise after:
First, I can collect all changes in a dict and make one update in the end. I will have more code here, And I will need to create an additional dict and then itterate over it. It won't be as efficient as the code above.
Second, I can update everytime I check an argument - this will create many queries to the database If I want to change many fields.. (1 query per field). I thinks it's also innefficient.

If I'm wrong, looking forward to read from you.)

Reply:
I just want you try to use update method, because this also an way to learn it better)
Collect all changes